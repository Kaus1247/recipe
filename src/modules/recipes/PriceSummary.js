import React from 'react';
import Box from '../../components/Box';
import { Row, Col } from '../../components/Grid';
import Flex from '../../components/Flex';
import List, { ListItem } from '../../components/List';
import Text from '../../components/Text';
import { parseRawPrice } from './price';

// Create PriceSummary user interface
const PriceSummary = ({ summary = (new Map()), totalPrice="$0.00", shippingPrice=0 }) => {
  return (
    <Box width={['290px', '450px']}>
      {
        summary.size > 0
          ? <List padding="16px">
              {
                Array.from(summary.values()).map((recipe, index)=>{
                  return (
                    <ListItem key={index+recipe.id} padding="0px 0px 8px 0px">
                      <Row>
                        <Col sm={9}>
                          <Flex alignItems="center" justifyContent="flex-start">
                            <Text fontWeight="400"> { recipe.name } x { recipe.selected }</Text>
                          </Flex>
                        </Col>
                        <Col sm={3}>
                          <Flex alignItems="center" justifyContent="flex-end">
                            <Text fontWeight="400">{ parseRawPrice(recipe.totalRecipePrice) }</Text>
                          </Flex>
                        </Col>
                      </Row>
                    </ListItem>
                  )
                })
              }
              <ListItem key={"shippingPrice"} padding="0px 0px 8px 0px">
                <Row>
                  <Col sm={9}>
                    <Flex alignItems="center" justifyContent="flex-start">
                      <Text fontWeight="400"> Shipping </Text>
                    </Flex>
                  </Col>
                  <Col sm={3}>
                    <Flex alignItems="center" justifyContent="flex-end">
                      <Text fontWeight="400">{ parseRawPrice(shippingPrice) }</Text>
                    </Flex>
                  </Col>
                </Row>
              </ListItem>
              <ListItem key={"totalBoxPrice"} padding="8px 0px 0px 0px" borderTop="1px solid #E4E4E4">
                <Row>
                  <Col sm={9}>
                    <Flex alignItems="center" justifyContent="flex-start">
                      <Text fontWeight="600"> Total </Text>
                    </Flex>
                  </Col>
                  <Col sm={3}>
                    <Flex alignItems="center" justifyContent="flex-end">
                      <Text fontWeight="600">{ totalPrice }</Text>
                    </Flex>
                  </Col>
                </Row>
              </ListItem>
            </List>
          : null
      }

    </Box>
  )
};

export default PriceSummary;
