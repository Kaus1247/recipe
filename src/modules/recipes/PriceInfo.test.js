import React from "react";
import PriceInfo from "./PriceInfo";
import { shallow } from "enzyme";
import Tooltip, { TooltipContainer } from "../../components/Tooltip";
import IconButton from "../../components/IconButton";
import IconInfoCircle from "../../icons/IconInfoCircle";
import PriceSummary from "./PriceSummary";

describe('PriceInfo', () => {
  it('renders without crash', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    expect(priceInfoWrapper).not.toEqual(null)
  })

  it('renders a tooltip container', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    const toolTipContainer = priceInfoWrapper.find(TooltipContainer);
    expect(toolTipContainer).toHaveLength(1);
  })

  it('renders a IconButton', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    const iconButton = priceInfoWrapper.find(IconButton);
    expect(iconButton).toHaveLength(1);
  })

  it('renders a IconInfoCircle', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    const iconInfoCircle = priceInfoWrapper.find(IconInfoCircle);
    expect(iconInfoCircle).toHaveLength(1);
  })

  it('does not renders a Tooltip initially', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    const tooltip = priceInfoWrapper.find(Tooltip);
    expect(tooltip).toHaveLength(0);
  })

  it('renders a Tooltip on click of icon button', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    const iconButton = priceInfoWrapper.find(IconButton);
    iconButton.simulate('click')
    const tooltip = priceInfoWrapper.find(Tooltip);
    expect(tooltip).toHaveLength(1);
  })

  it('renders a PriceSummary inside tooltip on click of icon button', () => {
    const priceInfoWrapper = shallow(<PriceInfo/>);
    const iconButton = priceInfoWrapper.find(IconButton);
    iconButton.simulate('click')
    const tooltip = priceInfoWrapper.find(Tooltip);
    const priceSummary = tooltip.find(PriceSummary)
    expect(priceSummary).toHaveLength(1);
  })

  it('passes all the of its own props to PriceSummary', () => {
    const summary = new Map();
    summary.set("recipeId1",{id: "recipeId1" , name : "recipe 1", selected : 2, totalRecipePrice: 1798 })
    const totalPrice = "$27.98";
    const shippingPrice = 1000;
    const priceInfoWrapper = shallow(<PriceInfo summary={summary} totalPrice={totalPrice} shippingPrice={shippingPrice}/>);
    const iconButton = priceInfoWrapper.find(IconButton);
    iconButton.simulate('click')
    const tooltip = priceInfoWrapper.find(Tooltip);
    const priceSummary = tooltip.find(PriceSummary)
    expect(priceSummary.props().summary).toEqual(summary);
    expect(priceSummary.props().totalPrice).toEqual(totalPrice);
    expect(priceSummary.props().shippingPrice).toEqual(shippingPrice);
  })

  it('should render correct layout before click on icon', () => {
    const summary = new Map();
    summary.set("recipeId1",{id: "recipeId1" , name : "recipe 1", selected : 2, totalRecipePrice: 1798 })
    const totalPrice = "$27.98";
    const shippingPrice = 1000;
    const priceInfoWrapper = shallow(<PriceInfo summary={summary} totalPrice={totalPrice} shippingPrice={shippingPrice}/>);
    expect(priceInfoWrapper.getElements()).toMatchSnapshot()
  })

  it('should render correct layout after click on icon', () => {
    const summary = new Map();
    summary.set("recipeId1",{id: "recipeId1" , name : "recipe 1", selected : 2, totalRecipePrice: 1798 })
    const totalPrice = "$27.98";
    const shippingPrice = 1000;
    const priceInfoWrapper = shallow(<PriceInfo summary={summary} totalPrice={totalPrice} shippingPrice={shippingPrice}/>);
    const iconButton = priceInfoWrapper.find(IconButton);
    iconButton.simulate('click')
    expect(priceInfoWrapper.getElements()).toMatchSnapshot()
  })


});
