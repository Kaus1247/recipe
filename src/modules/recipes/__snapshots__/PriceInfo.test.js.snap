// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInfo should render correct layout after click on icon 1`] = `
Array [
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-label="openBoxSummary"
      onClick={[Function]}
    >
      <IconInfoCircle
        size="20"
      />
    </ForwardRef(styled.button)>
    <ForwardRef(Styled(Box))>
      <PriceSummary
        shippingPrice={1000}
        summary={
          Map {
            "recipeId1" => Object {
              "id": "recipeId1",
              "name": "recipe 1",
              "selected": 2,
              "totalRecipePrice": 1798,
            },
          }
        }
        totalPrice="$27.98"
      />
    </ForwardRef(Styled(Box))>
  </ForwardRef(styled.div)>,
]
`;

exports[`PriceInfo should render correct layout before click on icon 1`] = `
Array [
  <ForwardRef(styled.div)>
    <ForwardRef(styled.button)
      aria-label="openBoxSummary"
      onClick={[Function]}
    >
      <IconInfoCircle
        size="20"
      />
    </ForwardRef(styled.button)>
  </ForwardRef(styled.div)>,
]
`;
